src/ProjectOrigin.PedersenCommitment.Tests/SumProofTests.cs

```diff
@@ -0,0 +1,146 @@
+using ProjectOrigin.PedersenCommitment.Ristretto;
+using System.Text;
+
+namespace ProjectOrigin.PedersenCommitment.Tests;
+
+public class SumProofTests
+{
+
+    [Fact]
+    public void SanityCheck()
+    {
+        // check that a zero commit is the same as doing it manually
+        var pc_gens = Generator.Default;
+        var r = Scalar.Random();
+        var c0 = pc_gens.Commit(0, r);
+        var c1 = pc_gens.H() * r;
+        Assert.Equal(c0, c1);
+    }
+
+    [Fact]
+    public void Zero()
+    {
+        var pc_gens = Generator.Default;
+        var label = Encoding.UTF8.GetBytes("test");
+        var r = Scalar.Random();
+        var c = pc_gens.Commit(0, r);
+
+
+
+        var proof = ZeroProof.Prove(pc_gens, r, label);
+
+        Assert.True(proof.Verify(pc_gens, c, label));
+    }
+
+    [Fact]
+    public void Equal()
+    {
+        var pc_gens = Generator.Default;
+        var label = Encoding.UTF8.GetBytes("test");
+        var r0 = Scalar.Random();
+        var r1 = Scalar.Random();
+        var c0 = pc_gens.Commit(42, r0);
+        var c1 = pc_gens.Commit(42, r1);
+
+        var c = c0 - c1;
+        var r = r0 - r1;
+
+        var proof = ZeroProof.Prove(pc_gens, r, label);
```

Comment:
Now I understand the gymnastics from the usage point of view through the ristretto library - as always tests tell the story of usage (y)